### PR TITLE
Fixed compilation errors on VS without PCH

### PR DESCRIPTION
**Pull request by sharpneli**
#### Summary
Bugfixes "Fixed compilation errors on VS without PCH"

#### Purpose of change
To fix failing compilation in VS2019 using cmake generated solution file without precompiled headers

#### Describe the solution
Added #include <string> into files that were missing it, lack of them is causing the build to fail when precompiled headers are not used. In addition changed the to_string_clipped definition to match the declaration by removing inconsistency about what parameters are const.

#### Describe alternatives you've considered
I considered enabling precompiled headers to get it to compile but that's super hacky. Better to just follow C++ specification.

#### Testing
Compiled successfully, created a characted that died horribly to wasps.

### CODE REVIEW

File: src/gamemode.h [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
 #include <iosfwd>
 #include <memory>
+#include <string>
```
Comment (akrieger) [inline src/gamemode.h:7]:
```suggestion
#include <iosfwd>
#include <string>
#include <memory>
```

**Review by akrieger — COMMENTED** [review]
Please keep includes sorted, otherwise LGTM.

File: src/calendar.h [hunk 2]
```diff
@@ -5,6 +5,7 @@
 #include <iosfwd>
 #include <utility>
 #include <vector>
+#include <string>
```
[the remark this answers is not in the document]
Comment (sharpneli) [reply src/calendar.h:8]: I don't have anything against this, but is there any particular reason for this ordering?

[the remark this answers is not in the document]
Comment (kevingranade) [reply src/calendar.h:8]: Just an arbitrary constraint to make things easier to read and keep them in a consistent order.  Also we're trying to get IWYU in a state where we can run it regularly, and it prefers to order the header inclusions for similar reasons.

File: src/calendar.h [hunk 3]
```diff
@@ -416,7 +417,7 @@ std::pair<int, clipped_unit> clipped_time( const time_duration &d );
  * 59 minutes will return "59 minutes".
  * @param align none, right, or compact.
  */
-std::string to_string_clipped( const time_duration &d, clipped_align align = clipped_align::none );
+std::string to_string_clipped( const time_duration &d, const clipped_align align = clipped_align::none );
```
Comment (akrieger) [inline src/calendar.h:420]: astyle wants to wrap this now.

```suggestion
std::string to_string_clipped( const time_duration &d,
                               const clipped_align align = clipped_align::none );
```

**Review by akrieger — APPROVED** [review]

**Pull request merged**